lib/fluent/plugin/in_s3.rb

```diff
@@ -42,6 +42,20 @@ def initialize
       desc "A unique identifier that is used by third parties when assuming roles in their customers' accounts."
       config_param :external_id, :string, default: nil
     end
+    # See the following link for additional params that could be added:
+    # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/STS/Client.html#assume_role_with_web_identity-instance_method
+    config_section :web_identity_credentials, multi: false do
+      desc "The Amazon Resource Name (ARN) of the role to assume"
+      config_param :role_arn, :string # required
```

Comment:
You can use `required: true` here like https://github.com/fluent/fluent-plugin-s3/pull/307/files#diff-f4e506e1e7bca94692f1db4ae6dd4716R94.

Reply:
Are you sure? The fluentd docs suggest the lack of `default` indicates required for the `config_param` bits.

https://docs.fluentd.org/plugin-development/api-config-types

Excerpt:

```ruby
# Required parameter. The configuration must have this parameter like 'param1 10'
config_param :param1, :integer
# Optional parameter. If the configuration doesn't have 'param2', 100 is used
config_param :param2, :integer, default: 100
```

Also doesn't seem to be reflected upstream: 

https://github.com/fluent/fluentd/blob/d9533da0b41fc652c51d86bf773db6a0075d7973/lib/fluent/config/configure_proxy.rb#L270-L283

That said, I think I set a default for the required value `web_identity_token_file`. I'll push an update in a few.